src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_lambda.json

```
@@ -22,5 +22,66 @@
       "NumberMax": 10000,
       "NumberMin": 1
     }
+  },
+  {
+    "op": "add",
+    "path": "/ValueTypes/AWS::Lambda::EventSourceMapping.EventSourceArn",
+    "value": {
+      "GetAtt": {
+        "AWS::DynamoDB::Table": "StreamArn",
+        "AWS::Kinesis::Stream": "Arn",
+        "AWS::SQS::Queue": "Arn"
+      },
+      "Ref": {
+        "Parameters": [
+          "String"
+        ],
+        "Resources": [
+          "AWS::MSK::Cluster"
+        ]
+      }
+    }
+  },
+  {
+    "op": "add",
+    "path": "/ValueTypes/AWS::Lambda::EventSourceMapping.StartingPosition",
+    "value": {
+      "AllowedValues": [
+        "LATEST",
+        "TRIM_HORIZON"
+      ]
+    }
```

could grab these [`AllowedValues` from `botocore`](https://github.com/aws-cloudformation/cfn-python-lint/pull/1682)
```bash
botocore $ git grep -C 4 -h 'TRIM_HORIZON' botocore/data/lambda/
    "EventSourcePosition":{
      "type":"string",
      "enum":[
        "TRIM_HORIZON",
        "LATEST",
        "AT_TIMESTAMP"
      ]
    },
```

```suggestion
    "value": {
      "botocore": "lambda/2015-03-31/EventSourcePosition"
    }
```

---

agreed.  Resolved.

---

We will include it but worth noting.  That requires an additional property that isn't supported by CloudFormation.